save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/TopBar.kt

```diff
@@ -43,271 +44,284 @@ import kotlinx.coroutines.cancel
             }
-            li {
-                className = ClassName("nav-item")
-                a {
-                    val hrefAnchor = FrontendRoutes.SANDBOX.path
-                    className = ClassName("nav-link d-flex align-items-center me-2 ${textColor(hrefAnchor, location)} active")
-                    style = jso {
-                        width = 9.rem
-                    }
-                    href = "#/$hrefAnchor"
-                    +"Try SAVE format"
+        }
+        li {
+            className = ClassName("nav-item")
+            a {
+                val hrefAnchor = FrontendRoutes.SANDBOX.path
+                className = ClassName(
+                    "nav-link d-flex align-items-center me-2 ${
+                        textColor(
+                            hrefAnchor,
+                            props.location
+                        )
+                    } active"
+                )
+                style = jso {
+                    width = 9.rem
                 }
+                href = "#/$hrefAnchor"
+                +"Try SAVE format"
             }
-            li {
-                className = ClassName("nav-item")
-                a {
-                    className = ClassName("nav-link d-flex align-items-center me-2 active")
-                    style = jso {
-                        width = 9.rem
-                    }
-                    href = "https://github.com/saveourtool/save-cloud"
-                    +"SAVE on GitHub"
+        }
+        li {
+            className = ClassName("nav-item")
+            a {
+                className = ClassName("nav-link d-flex align-items-center me-2 active")
+                style = jso {
+                    width = 9.rem
                 }
+                href = "https://github.com/saveourtool/save-cloud"
+                +"SAVE on GitHub"
             }
-            li {
-                className = ClassName("nav-item")
-                a {
-                    val hrefAnchor = FrontendRoutes.PROJECTS.path
-                    className = ClassName("nav-link d-flex align-items-center me-2 ${textColor(hrefAnchor, location)} active ")
-                    style = jso {
-                        width = 8.rem
-                    }
-                    href = "#/$hrefAnchor"
-                    +"Projects board"
+        }
+        li {
+            className = ClassName("nav-item")
+            a {
+                val hrefAnchor = FrontendRoutes.PROJECTS.path
+                className = ClassName(
+                    "nav-link d-flex align-items-center me-2 ${
+                        textColor(
+                            hrefAnchor,
+                            props.location
+                        )
+                    } active "
+                )
+                style = jso {
+                    width = 8.rem
                 }
+                href = "#/$hrefAnchor"
+                +"Projects board"
             }
-            li {
```

Comment:
What a bullshit you made again? You should have split the long method and have moved these `li` code blocks to separate methods:

```
        li {
            className = ClassName("nav-item")
            a {
                val hrefAnchor = FrontendRoutes.CONTESTS.path
                className = ClassName(
                    "nav-link d-flex align-items-center me-2 ${
                        textColor(
                            hrefAnchor,
                            props.location
                        )
                    } active"
                )
                style = jso {
                    width = 6.rem
                }
                href = "#/$hrefAnchor"
                +"Contests"
            }
        }
```

Reply:
Done in [a65257a](https://github.com/saveourtool/save-cloud/pull/1540/commits/a65257a221b20fe12514589ae83ddc1cbeccb0f4)